src/LanguageServer.Common/Utilities/DotnetInfo.cs

```diff
@@ -0,0 +1,835 @@
+    /// </summary>
+    /// <param name="Version">
+    ///     The SDK version.
+    /// </param>
+    /// <param name="BaseDirectory">
+    ///     The SDK base directory.
+    /// </param>
+    public record DotnetSdkInfo(SemanticVersion Version, string BaseDirectory)
+    {
+        /// <summary>
+        ///     Empty <see cref="DotnetRuntimeInfo"/>.
+        /// </summary>
+        public static readonly DotnetSdkInfo Empty = new DotnetSdkInfo(Version: null, BaseDirectory: null);
+    };
+
+    /// <summary>
+    ///     Information about a discovered .NET runtime.
+    /// </summary>
+    /// <param name="Name">
+    ///     The runtime name (e.g. "Microsoft.NETCore.App", "Microsoft.WindowsDesktop.App", "Microsoft.AspNetCore.App", etc).
+    /// </param>
+    /// <param name="Version">
+    ///     The SDK version.
+    /// </param>
+    public record DotnetRuntimeInfo(string Name, SemanticVersion Version)
+    {
+        /// <summary>
+        ///     Empty <see cref="DotnetRuntimeInfo"/>.
+        /// </summary>
+        public static readonly DotnetRuntimeInfo Empty = new DotnetRuntimeInfo(Name: null, Version: null);
+    };
+
+    /// <summary>
+    ///     Well-known names for various .NET runtimes.
+    /// </summary>
+    public static class WellKnownDotnetRuntimes
+    {
+        /// <summary>
+        ///     The name of the .NET runtime for ASP.NET.
+        /// </summary>
+        public static readonly string AspNetCore = "Microsoft.AspNetCore.App";
+
+        /// <summary>
+        ///     The name of the (default) .NET / .NET Core runtime.
+        /// </summary>
+        public static readonly string NetCore = "Microsoft.NETCore.App";
+
+        /// <summary>
+        ///     The name of the .NET runtime for Windows Desktop.
+        /// </summary>
+        public static readonly string WindowsDesktop = "Microsoft.WindowsDesktop.App";
+    }
+
+    /// <summary>
+    ///     Well-known process exit codes for the "dotnet" executable.
+    /// </summary>
+    /// <remarks>
+    ///     Derived from <seealso href="https://github.com/dotnet/runtime/blob/d123560a23078989f9563b83fa49a24802e88378/docs/design/features/host-error-codes.md"/>.
+    /// </remarks>
+    public enum DotNetHostExitCode
+        : int
+    {
+        /// <summary>
+        ///     Operation was successful.
+        /// </summary>
+        Success = 0,
+
+        /// <summary>
+        ///     Initialization was successful, but another host context is already initialized, so the returned context is "secondary" (the requested context was otherwise fully compatible with the already initialized context).
+        /// </summary>
+        /// <remarks>
+        ///     Probably not used as an actual exit code; returned by <c>hostfxr_initialize_for_runtime_config</c> if it's called when the host is already initialized in the process.
+        /// </remarks>
+        Success_HostAlreadyInitialized = 0x00000001,
```

Comment:
I am not sure how I feel about adding _all_ possible exit codes to this enum. Realistically we won't ever need most of them. Given that you've specified a place where we can get them in doc comment, I would probably follow YAGNI principle and remove everything we don't currently use.

Reply:
We may not need them, but it's useful to have a name for the exit code, if all we get is a screenshot of a log message (this happens quite frequently)...

Reply:
I'll remove the more esoteric ones (i.e. ones that should not realistically ever be returned to us by the .NET host process but I believe that the others are likely to be helpful in diagnosing failure-to-launch issues (e.g. even if we didn't know that the issue this PR relates to was about being unable to find a valid SDK, seeing `LibHostSdkFindFailure` in the log would have saved us a fair bit of time in terms of troubleshooting (as well as googling for related issues).

Reply:
I've removed any exit codes that seem unlikely to ever be returned in the scenarios where we execute the .NET host process.